### PR TITLE
feat(types): add vuex definition

### PR DESCRIPTION
**Pull request by orblazer**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

This add types definitions for `vuex`. With that we now have types in Actions/Mutations on :
- [`this.app`](https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-app/template/index.js#L105)
- [`this.$router`](https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-app/template/index.js#L52)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Comment by kevinmarrec** [comment]
@orblazer Thanks for the PR

I've found https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-app/template/index.js#L105 for `store.app` & https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-app/template/index.js#L52 for `store.$router` but I couldn't find for `store.route`, maybe you found it somewhere in documentation ?

Maybe another `store`props added by Nuxt would need type definitions ? I would want to be sure we cover properties that are properly existing.

Also, for consistency with other typedefs, you can remove semicolons from your type definition (I may review all types code style later as i'm working on a dedicated `@nuxt/types` in association with externalization of TypeScript support #5858)

File: packages/vue-app/types/vuex.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import Store from 'vuex'
+import Vue from 'vue'
+import VueRouter from 'vue-router'
+
+declare module 'vuex/types/index' {
+  interface Store<S> {
+    app: Vue;
```
Comment (kevinmarrec) [inline packages/vue-app/types/vuex.d.ts]:
```suggestion
    app: Vue
```

File: packages/vue-app/types/vuex.d.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    $router: VueRouter;
```
Comment (kevinmarrec) [inline packages/vue-app/types/vuex.d.ts]:
```suggestion
    $router: VueRouter
```

**Review by kevinmarrec — CHANGES_REQUESTED** [review]

**Review by kevinmarrec — CHANGES_REQUESTED** [review]
You also need to update `types/index.d.ts` and add

```js
// augment typings of Vuex
import './vuex'
```
after
```js
import './vue'
```

😜

**Comment by kevinmarrec** [comment]
Thanks for quick updates @orblazer , do you know if there's any other prop we may miss ?
Either way, looks good to me

**Comment by orblazer** [comment]
For Vuex types its goof for me too @kevinmarrec

**Review by kevinmarrec — CHANGES_REQUESTED** [review]
Hum, after some thoughts, `app` should be of type `NuxtAppOptions`, see https://github.com/nuxt/nuxt.js/blob/dev/packages/vue-app/types/index.d.ts#L17

can be imported like that : `import { NuxtAppOptions } from '.'` or `import { NuxtAppOptions } from './index'`

**Comment by orblazer** [comment]
Yes i have just realized that ;)

**Review by kevinmarrec — APPROVED** [review]

**Pull request merged**